### PR TITLE
fix(web): site details to display neighbour addresses

### PR DESCRIPTION
**Pull request by james-hill-kc**
## Describe your changes

update to book site visit page (site details to display neighbour addresses)

## Issue ticket number and link

BOAT-489

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.mapper.js [hunk 1]
```diff
@@ -15,3 +17,53 @@ export function mapWebVisitTypeToApiVisitType(webVisitType) {
...
 }
+
+/**
+ *
+ * @param {string} siteAddress
+ * @param {(string|null)[]} healthAndSafetyIssues
+ * @param {import('../appeal-details.types.js').AppealNeighbouringSite} neighbouringSiteInfo
+ * @returns {SummaryListRowProperties[]}
+ */
+export function buildSiteDetailsSummaryListRows(
```
Comment (AlexMoulin) [inline appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.mapper.js:28]: (data, currentRoute, session) as parameters for buildSiteDetailsSummaryListRows

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.mapper.js [hunk 2]
```diff
@@ -15,3 +17,53 @@ export function mapWebVisitTypeToApiVisitType(webVisitType) {
...
+	siteAddress,
+	healthAndSafetyIssues,
+	neighbouringSiteInfo
+) {
+	const rows = [
```
Comment (AlexMoulin) [inline appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.mapper.js:33]:
```
const mappedData = await initialiseAndMapAppealData(data, currentRoute, session);

let neighbouringSitesSummaryLists = [];
	if (data.appeal.neighbouringSite.contacts && data.appeal.neighbouringSite.contacts.length > 0) {
		for (const site of mappedData.appeal.neighbouringSite) {
			neighbouringSitesSummaryLists.push(site.display.summaryListItem);
		}
	}

const rows = [
mappedData.appeal.siteAddress.display.summaryListItem,
mappedData.appeal.lpaHealthAndSafety.display.summaryListItem,
mappedData.appeal.appellantHealthAndSafety.display.summaryListItem,
...neighbouringSitesSummaryLists
]

rows.forEach((row) => removeActions(row))

return rows

```

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js [hunk 3]
```diff
@@ -48,12 +51,18 @@ const renderScheduleSiteVisit = async (request, response) => {
 			healthAndSafetyIssues.push(appealDetails.healthAndSafety?.lpaQuestionnaire?.details);
 		}
 
+		const siteDetailsRows = buildSiteDetailsSummaryListRows(
+			formattedSiteAddress,
```
Comment (AlexMoulin) [inline appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js:55]: replace parameters with (data, currentUrl, session)

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js [hunk 4]
```diff
@@ -48,12 +51,18 @@ const renderScheduleSiteVisit = async (request, response) => {
 			healthAndSafetyIssues.push(appealDetails.healthAndSafety?.lpaQuestionnaire?.details);
 		}
```
Comment (AlexMoulin) [inline appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.controller.js:53]: const session = request.session;
const currentUrl = request.originalUrl;

**Review by AlexMoulin — COMMENTED** [review]
There I think I covered how to add display components using the mapper, let me know if something is amiss :)

File: appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.mapper.js (hunk 2 above)
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-details/site-visit/site-visit.mapper.js:32]: Thanks Alex, I've implemented these changes now :)

**Review by kandcdev — APPROVED** [review]

**Pull request merged**